src/main/java/net/nuagenetworks/bambou/ssl/DynamicKeystoreGenerator.java

```diff
@@ -0,0 +1,156 @@
+	private static final String KEYSTORE_INSTANCE_TYPE = "JKS";
+	private static final String CERTIFICATE_FACTORY_INSTANCE_TYPE = "X.509";
+	private static final String KEY_CHARSET = "UTF-8";
+	private static final String PRIVATE_KEY_HEADER_TRAILER_REGEX = "(-+BEGIN PRIVATE KEY-+\\r?\\n|-+END PRIVATE KEY-+\\r?\\n?)";
+    
+    private DynamicKeystoreGenerator() {}
+	
+	public static KeyManager[] generateKeyManagersForCertificates(String[] certificateFilePairPaths) throws KeyManagementException {
+        logger.debug("Creating a key manager for files : " + certificateFilePairPaths[0] + " and " + certificateFilePairPaths[1]);
+		try {
+	        String pathToCertificatePEMFile = certificateFilePairPaths[0];
+	        String pathToPrivateKeyPEMFile = certificateFilePairPaths[1];
+			KeyManager[] keyManagers = {};
+			Certificate certificate = getCertificate(pathToCertificatePEMFile);
+			
+			String keyFileContents = getContentsOfPrivateKeyPEMFile(pathToPrivateKeyPEMFile);
+			byte[] decodedPrivateKey = getDecodedPrivateKey(keyFileContents);
+
+			RSAPrivateKey rsaPrivateKey = generateRSAPrivateKey(decodedPrivateKey);
+
+			KeyStore keystore = createAndLoadDynamicKeystore(rsaPrivateKey, certificate);
+			keyManagers = generateKeyManagersForDynamicKeystore(keystore);
+			return keyManagers;
+		} catch (Exception e) {
+			throw new KeyManagementException(e);
+		}
+	}
+
+	public static Certificate getCertificate(String certificatePEMFile) throws KeyManagementException {
+        logger.debug("Creating a Certificate for file : " + certificatePEMFile);
+		Certificate certificate = null;
+		try (FileInputStream fis = new FileInputStream(certificatePEMFile)) {
+			certificate = generateX509Certificate(fis);
+		} catch (IOException e) {
+			throw new KeyManagementException(e);
+		}
+	    return certificate;
+	}
+	
+	public static Certificate generateX509Certificate(InputStream is) throws KeyManagementException {
+        logger.debug("Generating X509 Certificate");
+		Certificate certificate = null;
+		try {
+			CertificateFactory cf = CertificateFactory.getInstance(CERTIFICATE_FACTORY_INSTANCE_TYPE);
+			certificate = cf.generateCertificate(is);
+		} catch (CertificateException e) {
+			throw new KeyManagementException(e);
+		}
+	    return certificate;
+	}
+	
+	public static String getContentsOfPrivateKeyPEMFile(String privateKeyPEMFile) throws KeyManagementException {
+        logger.debug("Reading key contents for file : " + privateKeyPEMFile);
+	    byte[] keyFileBytes = {};
+	    String privateKeyContents = "";
+		try {
+			Path path = Paths.get(privateKeyPEMFile);
+			keyFileBytes = Files.readAllBytes(path);
+			privateKeyContents = new String(keyFileBytes, KEY_CHARSET);
+		} catch (IOException e) {
+			throw new KeyManagementException(e);
+		}
+		return privateKeyContents;
+	}
+	
+	public static byte[] getDecodedPrivateKey(String keyFileContents) {
+        logger.debug("Decoding private key : " + keyFileContents);
```

Comment:
Indentation issue.  Make sure to use spaces instead of tabs. Applies to all changes.